zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZNodeGroupAclUtil.java

```diff
@@ -33,4 +37,24 @@
   // Has the same effect as the ZK client using ZooDefs.Ids.CREATOR_ALL_ACL.
   public static final String SET_X509_CLIENT_ID_AS_ACL =
       ZNODE_GROUP_ACL_CONFIG_PREFIX + "setX509ClientIdAsAcl";
+  // A list of domain names that will have super user privilege, separated by ","
+  public static final String SUPER_USER_DOMAIN_NAME =
+      ZNODE_GROUP_ACL_CONFIG_PREFIX + "superUserDomainName";
+  // A list of znode path prefixes, separated by ","
+  // Znode whose path starts with the defined path prefix would have open read access
+  // Meaning the znode will have (world:anyone, r) ACL
+  public static final String OPEN_READ_ACCESS_PATH_PREFIX =
+      ZNODE_GROUP_ACL_CONFIG_PREFIX + "openReadAccessPathPrefix";
+
+  public static String[] getOpenReadAccessPathPrefixes() {
+    String openReadAccessPathPrefixesStr = System.getProperty(OPEN_READ_ACCESS_PATH_PREFIX, "");
+    return openReadAccessPathPrefixesStr == null ? new String[0] : openReadAccessPathPrefixesStr.split(",");
```

Comment:
I commented on why this wasn't returning a `Set<String>` instead - if you make it a set, you could use a constant `Collections.emptySet()` and not incur any memory cost. If you prefer returning a string array, I suggest using a string array constant to cut down on memory usage/GC? See https://stackoverflow.com/questions/27167199/is-it-worth-to-make-a-constant-for-an-empty-array-in-java for an example.

Reply:
`I commented on why this wasn't returning a Set<String> instead` - I explained that the usage patterns are different so I used an array here.
But your point is valid, and on the second thought, with all the considerations I made it to return a set instead of an array.